gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/linker/JmsDestinationLinker.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2018. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayexport.tasks.explode.linker;
+
+import com.ca.apim.gateway.cagatewayconfig.beans.Bundle;
+import com.ca.apim.gateway.cagatewayconfig.beans.JmsDestination;
+import com.ca.apim.gateway.cagatewayconfig.beans.Service;
+import com.ca.apim.gateway.cagatewayconfig.beans.StoredPassword;
+import org.jetbrains.annotations.NotNull;
+
+import javax.inject.Singleton;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.ca.apim.gateway.cagatewayconfig.util.gateway.VariableUtils.extractVariableName;
+import static com.ca.apim.gateway.cagatewayexport.tasks.explode.linker.ServiceLinker.getServicePath;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@Singleton
+public class JmsDestinationLinker implements EntityLinker<JmsDestination> {
+
+    private static final Pattern secpassPattern = Pattern.compile("secpass.(.+?).plaintext");
+    
+    @Override
+    public Class<JmsDestination> getEntityClass() {
+        return JmsDestination.class;
+    }
+
+    @Override
+    public void link(JmsDestination entity, Bundle bundle, Bundle targetBundle) {
+        linkJndiStoredPassword(entity, bundle);
+        linkDestinationStoredPassword(entity, bundle);
+        linkInboundAssociatedService(entity, bundle);
+        
+        // (kpak) - link private key(s)
```

Comment:
The plugin is supporting only GENERIC KeyStore, which is the filesystem ks.
The `KeyStoreType` class has the constants for all but we use only generic.

The linking only need to check if the key alias are avaliable in the private key entity map, nothing more than that.

To make the whole thing work you might want to create fields in the json for each key, containing only the key alias, to be saved in the yaml/json. Then when rebuilding the bundle you pick this alias and use the `getId` from the `KeyStoreType.GENERIC` to build the whole thing.

Reply:
Done.